R/core_functions.R

```diff
@@ -64,21 +63,19 @@ get_pred <- function(
     time, pred, is_delay, population, developing_population, max_delay,
     predictors
   ) {
+
   life_stages <- rownames(population)
 
-  if (pred %in% valid_predictors_from_table(predictors)) {
-    # TODO "host_den" is hardcoded here as the only predictor from the predictors
-    # table for which we only use the predictor value at the first day of the
-    # transition. Should this be part of the configuration for each predictor?
-    if (is_delay && pred != "host_den") {
+  if (pred$pred %in% valid_predictors_from_table(predictors)) {
+    if (is_delay && !pred$first_day_only) {
```

Comment:
This is the main behavior change in the actual modeling code. We use the `first_day_only` flag rather than whether the pred is `"host_den"` to determine whether to make `time` a vector/scalar

Reply:
Good catch that this was hard-coded in for host_den!!